test/controllers/web-payments/google-pay/normalise-google-pay-payload.test.js

```diff
@@ -0,0 +1,80 @@
+const expect = require('chai').expect
+const normalise = require('../../../../app/controllers/web-payments/google-pay/normalise-google-pay-payload')
+
+describe('normalise Google Pay payload', () => {
+  it('should return the correct format for the payload', () => {
+    const googlePayPayload = {
+      details: {
+        apiVersionMinor: 0,
+        apiVersion: 2,
+        paymentMethodData: {
+          description: 'Mastercard •••• 4242',
+          info: {
+            cardNetwork: 'MASTERCARD',
+            cardDetails: '4242'
+          },
+          tokenizationData: {
+            type: 'PAYMENT_GATEWAY',
+            token: '{"signature":"MEQCIB54h8T/hWY3864Ufkwo4SF5IjhoMV9hjpJRIsqbAn4LAiBZz1VBZ+aiaduX8MN3dBtzyDOZVstwG/8bqJZDbrhKfQ\\u003d","protocolVersion":"ECv1","signedMessage":"{\\"encryptedMessage\\":\\"I2fg4rbEzgRHpvJqN4pHa7Zh93eGePLCKwHljtTfgWELsLOx0Or1cBQ6giKNUrJza3DqhS0AO+Qoar44J2HBryMRaiuSIi/un0zsNV9cfmiLSHNjHNnATkYrfY4u/Or2uSeuAaxLEt3d91NhHWKqf6BelNme182onG23GvOqd4RAukUW7RJ04eouaCIvBisdt866uq/9B3jJb0QiT91ifZ8C/bfScnBFPL4AX0X3G+B7418wSTtUYrMVBnyhNJS8T2Aw9oW8s7c9pra4PI9cHfcu22opRxzSS9snBF39uTwk8c+Pjj3G8yfNI0biDkHNAUiA1YuBW5CgHeJZ/FhayAsAPzfMmI4qei9nTzIEPlDkkGsqFnamCYIg8N9SM51YV8NGS0MQTIYmJg3GHl2D/We30D6dvYSWfLDDJNATAgb3l+4powoxogb28cwE9vLjFhOF/GChrGRpM845E9r1q0tNfg\\\\u003d\\\\u003d\\",\\"ephemeralPublicKey\\":\\"BC9B7aoVhvPzR54m8P1CkGFDSyuxl04iqu22SvnrlLgZEoH3EvpBNKPyMS/nLIe3mJ+cw26GglqMs00B7EEEs0I\\\\u003d\\",\\"tag\\":\\"lOylJZZF3xdVpsqpl5bKgZdsxRPetMRvcKu9WnmFQ/k\\\\u003d\\"}"}'
+          },
+          type: 'CARD'
+        }
+      },
+      payerEmail: 'name@email.fyi',
+      payerName: 'Some Name'
+    }
+
+    const normalisedPayload = normalise(googlePayPayload)
+    expect(normalisedPayload).to.eql(
+      {
+        payment_info: {
+          last_digits_card_number: '4242',
+          brand: 'master-card',
+          cardholder_name: 'Some Name',
+          email: 'name@email.fyi'
+        },
+        encrypted_payment_data: {
+          signature: 'MEQCIB54h8T/hWY3864Ufkwo4SF5IjhoMV9hjpJRIsqbAn4LAiBZz1VBZ+aiaduX8MN3dBtzyDOZVstwG/8bqJZDbrhKfQ=',
+          protocol_version: 'ECv1',
+          signed_message: 'I2fg4rbEzgRHpvJqN4pHa7Zh93eGePLCKwHljtTfgWELsLOx0Or1cBQ6giKNUrJza3DqhS0AO+Qoar44J2HBryMRaiuSIi/un0zsNV9cfmiLSHNjHNnATkYrfY4u/Or2uSeuAaxLEt3d91NhHWKqf6BelNme182onG23GvOqd4RAukUW7RJ04eouaCIvBisdt866uq/9B3jJb0QiT91ifZ8C/bfScnBFPL4AX0X3G+B7418wSTtUYrMVBnyhNJS8T2Aw9oW8s7c9pra4PI9cHfcu22opRxzSS9snBF39uTwk8c+Pjj3G8yfNI0biDkHNAUiA1YuBW5CgHeJZ/FhayAsAPzfMmI4qei9nTzIEPlDkkGsqFnamCYIg8N9SM51YV8NGS0MQTIYmJg3GHl2D/We30D6dvYSWfLDDJNATAgb3l+4powoxogb28cwE9vLjFhOF/GChrGRpM845E9r1q0tNfg=='
```

Comment:
i think this signed_message is wrong. it literally needs to send in the stringified json payload. see https://github.com/alphagov/pay-connector/pull/1035/files -> GooglePayForWorldpayTest for example. the `signedMessage` there is a string containing a json.

Reply:
Thats the normalised payload, the one google send is just above on line 18

Reply:
ok but what do we do with this normalisedPayload? Do we send it to connector?

Reply:
Yeah it’s normalised to match the format that connector expects.

Reply:
Are you saying connector expects the original JSON-as-string version? By normalising it it looks exactly the same as Apple Pay’s token data, which I think is what @danworth based his work on, so I think it’s ok.

Reply:
It works end-to-end when I try it locally so surely I am sending connector the correct payload?

Reply:
Ah ok, so I should send it as a `string` and then connector will decode it, do we not want to do it the same as Apple Pay? It’s trivial either way I guess.

Reply:
no connector doesn't decode anything. it just passes that string through to worldpay. if you have a look at http://support.worldpay.com/support/kb/gg/corporate-gateway-guide/content/mobilewallets/googlepay.htm and see the "Example payload" you'll see it sends the json-as-string: ```<signedMessage>{"encryptedMessage":"qrnk5KIV1syX...```